build.ps1

```diff
@@ -0,0 +1,923 @@
+    -G Ninja `
+    -S $SourceCache\llvm-project\llvm
+}
+
+function Build-ZLib($Arch)
+{
+  $ArchName = $Arch.ShortName
+
+  Build-CMakeProject `
+    -Arch $Arch `
+    -B $BinaryCache\zlib-1.2.11.$ArchName `
+    -D BUILD_SHARED_LIBS=NO `
+    -D CMAKE_BUILD_TYPE=Release `
+    -D CMAKE_MT=mt `
+    -D CMAKE_INSTALL_PREFIX=$InstallRoot\zlib-1.2.11\usr `
+    -D INSTALL_BIN_DIR=$InstallRoot\zlib-1.2.11\usr\bin\$ArchName `
+    -D INSTALL_LIB_DIR=$InstallRoot\zlib-1.2.11\usr\lib\$ArchName `
+    -D SKIP_INSTALL_FILES=YES `
+    -G Ninja `
+    -S $SourceCache\zlib `
+    -BuildDefaultTarget `
+    -Install
+}
+
+function Build-XML2($Arch)
+{
+  $ArchName = $Arch.ShortName
+
+  Build-CMakeProject `
+    -Arch $Arch `
+    -B $BinaryCache\libxml2-2.9.12.$ArchName `
+    -D BUILD_SHARED_LIBS=NO `
+    -D CMAKE_BUILD_TYPE=Release `
+    -D CMAKE_MT=mt `
+    -D CMAKE_INSTALL_PREFIX=$InstallRoot\libxml2-2.9.12\usr `
+    -D CMAKE_INSTALL_BINDIR=bin/$ArchName `
+    -D CMAKE_INSTALL_LIBDIR=lib/$ArchName `
+    -D LIBXML2_WITH_ICONV=NO `
+    -D LIBXML2_WITH_ICU=NO `
+    -D LIBXML2_WITH_LZMA=NO `
+    -D LIBXML2_WITH_PYTHON=NO `
+    -D LIBXML2_WITH_TESTS=NO `
+    -D LIBXML2_WITH_THREADS=YES `
+    -D LIBXML2_WITH_ZLIB=NO `
+    -G Ninja `
+    -S $SourceCache\libxml2 `
+    -BuildDefaultTarget `
+    -Install
+}
+
+function Build-CURL($Arch)
+{
+  $ArchName = $Arch.ShortName
+
+  Build-CMakeProject `
+    -Arch $Arch `
+    -B $BinaryCache\curl-7.77.0.$ArchName `
+    -D BUILD_SHARED_LIBS=NO `
+    -D CMAKE_BUILD_TYPE=Release `
+    -D CMAKE_MT=mt `
+    -D CMAKE_INSTALL_PREFIX=$InstallRoot\curl-7.77.0\usr `
+    -D CMAKE_INSTALL_BINDIR=bin/$ArchName `
+    -D CMAKE_INSTALL_LIBDIR=lib/$ArchName `
+    -D BUILD_CURL_EXE=NO `
+    -D CMAKE_USE_OPENSSL=NO `
+    -D CURL_CA_PATH=none `
+    -D CMAKE_USE_SCHANNEL=YES `
+    -D CMAKE_USE_LIBSSH2=NO `
+    -D HAVE_POLL_FINE=NO `
+    -D CURL_DISABLE_LDAP=YES `
+    -D CURL_DISABLE_LDAPS=YES `
+    -D CURL_DISABLE_TELNET=YES `
+    -D CURL_DISABLE_DICT=YES `
+    -D CURL_DISABLE_FILE=YES `
+    -D CURL_DISABLE_TFTP=YES `
+    -D CURL_DISABLE_RTSP=YES `
+    -D CURL_DISABLE_PROXY=YES `
+    -D CURL_DISABLE_POP3=YES `
+    -D CURL_DISABLE_IMAP=YES `
+    -D CURL_DISABLE_SMTP=YES `
+    -D CURL_DISABLE_GOPHER=YES `
+    -D CURL_ZLIB=YES `
+    -D ENABLE_UNIX_SOCKETS=NO `
+    -D ENABLE_THREADED_RESOLVER=NO `
+    -D ZLIB_ROOT=$InstallRoot\zlib-1.2.11\usr `
+    -D ZLIB_LIBRARY=$InstallRoot\zlib-1.2.11\usr\lib\$ArchName\zlibstatic.lib `
+    -G Ninja `
+    -S $SourceCache\curl `
+    -BuildDefaultTarget `
+    -Install
+}
+
+function Build-ICU($Arch)
+{
+  $ArchName = $Arch.ShortName
+
+  if (-not(Test-Path -Path "$SourceCache\icu\icu4c\CMakeLists.txt"))
+  {
+    Copy-Item $SourceCache\swift-installer-scripts\shared\ICU\CMakeLists.txt $SourceCache\icu\icu4c\
+    Copy-Item $SourceCache\swift-installer-scripts\shared\ICU\icupkg.inc.cmake $SourceCache\icu\icu4c\
+  }
+
+  if ($Arch -eq $ArchARM64)
```

Comment:
We should compare this to `%PROCESSOR_ARCHITECTURE%`.  If we are building for a foreign architecture, we need to use the host tools.